docs/customization.rst

```diff
@@ -1,5 +1,185 @@
 :hero: Configuration options to personalize your site.
 
+.. embedded material icons used for inline demonstration
+.. raw:: html
+
+    <style>
```

Comment:
It would be good to figure out a better way to handle icons, so that we don't have to rely on raw html and hard coding the svg icons into the html.

Reply:
I could add a extra css for the docs, but I didn't want it used for all the pages. Also, I wanted the icons to change color with the different schemes. This was all I could find that would let me do that.

It is pretty annoying that the icons aren't copied with the theme's assets. (We're talking like less than 5 MB).

Reply:
I've found this extension called sphinx-panels which adds custom roles for [inline icons](https://sphinx-panels.readthedocs.io/en/latest/#inline-icons). Unfortunately, the material design icons aren't available. However, They are moving their code base to a newer extension called [sphinx-design](https://github.com/executablebooks/sphinx-design). I've raised an issue to see if inline material icon are on the horizon.

Reply:
What is the use case for referencing icons directly in the rst?  I see that in customization.rst they are used as examples for customizing this theme.  It seems like they would more be useful for customizing admonitions, or defining new directives, rather than being used directly.  Unfortunately pre-building the css makes it a bit tricky to change the icons properly, since they are inlined into the css.  On the other hand, if the css isn't pre-built then some nodejs dependencies are needed when building documentation, which would also be unfortunate.

Reply:
Instead of just showing the names of the icons, I wanted to show the icon itself (next to the icon's name). A picture speaks a thousand words...

Yes, I would like an easier way to customize the admonitions' icon. Sphinx also has a generic admonition that can be used to add new user-defined admonitions, but for some reason this theme treats generic admonitions like a `.. note::`.

I can imagine that some users might want to put them in the section headers (hopefully resulting in a icon in the nav menu items also).

Reply:
Yeah, though unfortunately libsass is deprecated --- I don't think it supports all of the sass syntax used by mkdocs-material, though I could be mistaken.  The recommended replacement is dart sass, but I don't think there are Python bindings for that.